src/ui/public/visualize/loader/embedded_visualize_handler.js

```diff
@@ -26,58 +30,144 @@ const RENDER_COMPLETE_EVENT = 'render_complete';
  * with the visualization.
  */
 export class EmbeddedVisualizeHandler {
-  constructor(element, scope, savedObject) {
+  constructor(element, savedObject, params) {
     this._element = element;
-    this._scope = scope;
     this._savedObject = savedObject;
+    this._params = {
+      uiState: params.uiState,
+      queryFilter: params.queryFilter,
+      searchSource: savedObject.searchSource,
+      aggs: savedObject.vis.getAggConfig(),
+      timeRange: params.timeRange,
+      filters: params.filters,
+      query: params.query,
+    };
+
     this._listeners = new EventEmitter();
     // Listen to the first RENDER_COMPLETE_EVENT to resolve this promise
     this._firstRenderComplete = new Promise(resolve => {
       this._listeners.once(RENDER_COMPLETE_EVENT, resolve);
     });
-    this._element.on('renderComplete', () => {
+    this._element.addEventListener('renderComplete', () => {
       this._listeners.emit(RENDER_COMPLETE_EVENT);
     });
+
+    this._init(params);
```

Comment:
Why not put all of the `init` logic in the constructor too? If we could do that, then we could destructure `appState` and `Private` as well, and reference them on lines 62 and 71. We could reference `vis` directly.

Here's how it would look (I grouped a couple related lines together slightly differently too):

```js
  constructor(element, savedObject, params) {
    this._element = element;
    this._savedObject = savedObject;

    const { searchSource, vis } = savedObject;

    const {
      appState,
      uiState,
      queryFilter,
      timeRange,
      filters,
      query,
      Private,
    } = params;

    this._params = {
      uiState: uiState,
      queryFilter: queryFilter,
      searchSource: searchSource,
      aggs: vis.getAggConfig(),
      timeRange: timeRange,
      filters: filters,
      query: query,
    };

    this._appState = appState;

    this._listeners = new EventEmitter();
    // Listen to the first RENDER_COMPLETE_EVENT to resolve this promise
    this._firstRenderComplete = new Promise(resolve => {
      this._listeners.once(RENDER_COMPLETE_EVENT, resolve);
    });
    this._element.addEventListener('renderComplete', () => {
      this._listeners.emit(RENDER_COMPLETE_EVENT);
    });

    this._loaded = false;
    this._destroyed = false;

    this._vis = vis;
    this._vis._setUiState(uiState);
    this._vis.on('update', this._handleVisUpdate);
    this._vis.on('reload', this._reloadVis);

    this._uiState = vis.getUiState();
    this._uiState.on('change', this._fetchAndRender);

    this._visualize = new VisualizeDataLoader(vis, Private);
    this._renderCompleteHelper = new RenderCompleteHelper(this._element);

    this._render();
  }
```